DQMOffline/Trigger/plugins/RazorMonitor.cc

```diff
@@ -0,0 +1,371 @@
+// -----------------------------
+// 
+// Offline DQM for razor triggers. The razor inclusive analysis measures trigger efficiency
+// in SingleElectron events (orthogonal to analysis), as a 2D function of the razor variables
+// M_R and R^2. Also monitor dPhi_R, used offline for  QCD and/or detector-related MET tail 
+// rejection.
+// Based on DQMOffline/Trigger/plugins/METMonitor.*
+//
+// -----------------------------
+
+
+#include "DQMOffline/Trigger/plugins/RazorMonitor.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+RazorMonitor::RazorMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_             ( iConfig.getParameter<std::string>("FolderName") )
+  , metToken_             ( consumes<reco::PFMETCollection>      (iConfig.getParameter<edm::InputTag>("met")       ) )   
+  , jetToken_             ( mayConsume<reco::PFJetCollection>      (iConfig.getParameter<edm::InputTag>("jets")      ) )   
+  , ls_binning_           ( getHistoLSPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("lsPSet")     ) )
+  , rsq_binning_          ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >   ("rsqBins")    )
+  , mr_binning_           ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >   ("mrBins")     )
+  , dphiR_binning_        ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >   ("dphiRBins")  )
+  , num_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("numGenericTriggerEventPSet"),consumesCollector(), *this))
+  , den_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("denGenericTriggerEventPSet"),consumesCollector(), *this))
+  , metSelection_ ( iConfig.getParameter<std::string>("metSelection") )
+  , jetSelection_ ( iConfig.getParameter<std::string>("jetSelection") )
+  , njets_      ( iConfig.getParameter<uint>("njets" )      )
+  , rsqCut_     ( iConfig.getParameter<double>("rsqCut" )  )
+  , mrCut_      ( iConfig.getParameter<double>("mrCut" )   )
+{
+
+  theHemispheres_ = consumes<std::vector<math::XYZTLorentzVector> >(iConfig.getParameter<edm::InputTag>("hemispheres"));
+
+  MR_ME_.numerator = nullptr;
+  MR_ME_.denominator = nullptr;
+  MRVsLS_.numerator = nullptr;
+  MRVsLS_.denominator = nullptr;
+  Rsq_ME_.numerator = nullptr;
+  Rsq_ME_.denominator = nullptr;
+  RsqVsLS_.numerator = nullptr;
```

Comment:
I think we should either monitor both, or neither, vs LS. They are sensitive to different things, so it's not clear to me we should favor one or the other, but on the other hand anything we see in MR or Rsq will also show up in HT / MET.